Gordon360/Services/RecIM/ParticipantService.cs

```diff
@@ -0,0 +1,242 @@
+                                    (pt, t) => new TeamExtendedViewModel
+                                    {
+                                        ID = t.ID,
+                                        Activity = _context.Activity.FirstOrDefault(a => a.ID == t.ActivityID),
+                                        Name = t.Name,
+                                        Status = _context.TeamStatus
+                                                    .FirstOrDefault(ts => ts.ID == t.StatusID)
+                                                    .Description,
+                                        Logo = t.Logo
+                                    });
+            return teams;
+        }
+
+        public IEnumerable<ParticipantExtendedViewModel> GetParticipants()
+        {
+            //temporary, slow and will be adjusted after we implement views in the DB
+            var participants = _context.Participant.Select(p => new ParticipantExtendedViewModel
+            {
+                Username = p.Username,
+                Email = _accountService.GetAccountByUsername(p.Username).Email,
+                Status = _context.ParticipantStatusHistory
+                                                .Where(psh => psh.ParticipantUsername == p.Username)
+                                                .OrderByDescending(psh => psh.ID)
+                                                .Take(1)
+                                                    .Join(_context.ParticipantStatus,
+                                                        psh => psh.StatusID,
+                                                        ps => ps.ID,
+                                                        (psh, ps) => ps.Description)
+                                                .FirstOrDefault(),
+                IsAdmin = p.IsAdmin
+            });
+            return participants;
+        }
+
+        public async Task<ParticipantExtendedViewModel> PostParticipantAsync(string username, int? statusID = 4)
+        {
+            await _context.Participant.AddAsync(new Participant
+            {
+                Username = username
+            });
+            await _context.ParticipantStatusHistory.AddAsync(new ParticipantStatusHistory
+            {
+                ParticipantUsername = username,
+                StatusID = statusID ?? 4, //default to cleared
+                StartDate = DateTime.Now,
+                //No defined end date for creation
+            });
+            await _context.SaveChangesAsync();
+            var participant = GetParticipantByUsername(username);
+            return participant;
+        }
+
+        public async Task<ParticipantExtendedViewModel> UpdateParticipantAsync(string username, bool isAdmin)
+        {
+            var participant = GetParticipantByUsername(username);
+            participant.IsAdmin = isAdmin;
+            await _context.SaveChangesAsync();
+            return participant;
```

Comment:
I think this may not work, because `participant` here is a ViewModel and is not being tracked by `_context`, so the change to the object does not change anything in the database I believe.

Reply:
Good point, this was definitely not checked since we haven't had a UI use case made yet. Regardless, it is now modified.